### PR TITLE
Add a content editor permission role (m2m with sites) and add permission check for content items

### PR DESCRIPTION
**Pull request by MyPyDavid**
In order to make content items (eg Catalogs etc..) editable only by users with editor roles from certain sites. 

An extra field  `content_editor` was added to the `Role` model.
An extra field to `editor_sites` was added to the `Catalog` model.
The "permission check" or validation is now done through the `LockedValidator`.

Draft for #490

### CODE REVIEW

File: rdmo/accounts/models.py [hunk 1]
```diff
@@ -138,6 +138,12 @@ class Role(models.Model):
         help_text=_('The sites for which this user is manager.')
     )
 
+    content_editor = models.ManyToManyField(
```
Comment (jochenklar) [inline rdmo/accounts/models.py:141]: I would just call it `editor`, all the fields are for `Site`.

**Comment by jochenklar** [comment]
I am wondering if, instead of adding `editor_sites` we should make `sites` a "through" model, and give it a `role`. This would be cleaner. Now you can have cases where a site could be "editor", but has no access.

**Comment by jochenklar** [comment]
I think enforcing permissions through the `LockedValidator` is not the way to go. In `projects` we already use rules for object based permissions, it is a bit more complicated, but cleaner I think.

File: rdmo/accounts/models.py (hunk 1 above)
Comment (MyPyDavid) [reply rdmo/accounts/models.py:141]: yes `editor` would be best.
Then for the special case where editor has "all Sites" for example call it the `instance_editor` property?
```py
@property
def instance_editor(self):
    return self.editor.all().count() == Site.objects.all().count()
```

**Comment by MyPyDavid** [comment]
thank you! Yes I saw the `HasObjectPermission` on the `projects` but was still looking for a way how to apply that to the `catalog` models.
We want to have similar permission functionality (read/write) as the `LockedValidator`  does but it (definitively) should not go through that necessarily in the final version of this PR. 
I have to look into the "through" option, also saw it somewhere else (`through='Membership'`). 
Through `role` we could give it the `editor` ?  But how can we then differentiate between read or write permissions?

**Comment by MyPyDavid** [comment]
I will split the features of this PR into separate PRs for adding the `Role.editor` field and adding permissions for `catalog` etc..

**Pull request closed without merging**